packages/camera/camera_android_camerax/lib/src/plane_proxy.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:flutter/services.dart' show BinaryMessenger;
+import 'package:meta/meta.dart' show protected;
+
+import 'android_camera_camerax_flutter_api_impls.dart';
+import 'camerax_library.g.dart';
+import 'instance_manager.dart';
+import 'java_object.dart';
+
+/// A single color plane of image data.
+///
+/// See https://developer.android.com/reference/androidx/camera/core/ImageProxy.PlaneProxy.
+class PlaneProxy extends JavaObject {
+  /// Constructs a [PlaneProxy] that is not automatically attached to a native object.
+  PlaneProxy.detached(
+      {BinaryMessenger? binaryMessenger, InstanceManager? instanceManager})
+      : super.detached(
+            binaryMessenger: binaryMessenger,
+            instanceManager: instanceManager) {
+    _api = _PlaneProxyHostApiImpl(
+        binaryMessenger: binaryMessenger, instanceManager: instanceManager);
+    AndroidCameraXCameraFlutterApis.instance.ensureSetUp();
+  }
+
+  late final _PlaneProxyHostApiImpl _api;
+
+  /// Returns the pixels buffer containing frame data.
+  Future<Uint8List> getBuffer() {
```

Comment:
I should convert these getters to class fields.

Reply:
I thought about the further, and I'm not sure that I should. Folks may want to get some data but not others, and it may make a difference if someone just wants the image width/height versus all of the planes from a particular frame and more. If folks disagree, maybe I can file an issue and keep this as is for now.

Reply:
I think `pixelStride` and `rowStride` could be fields since they aren't expensive to pass. The image buffer is, so I think this implementation is good. I think it's fine to keep as is for now and can be changed later.

Reply:
Can you lay out pros/cons or maybe use cases that are harder/easier for getters vs the current implementation.

Reply:
Actually, I just read https://github.com/flutter/packages/pull/3454/files#r1174443624. I think this could be a field as well since a person would already have to await `await ImageProxy.getPlanes()` and you wouldn't want to `await` twice. My guess would be that a user would want the byte data when they call `getPlanes` since `height`/`width` is already included in `ImageProxy`.

Reply:
I decided to go ahead and covert all of the getters in `ImageProxy` except for `ImageProxy.getPlanes()` to fields. My reasoning on that is that `ImageProxy.getPlanes()` is the most expensive operation, so if for some reason folks want other image information, but not the plane information, they don't have to perform it.

I converted all of the getters in `PlaneProxy` to fields because I think @bparrishMines is right about the fact that if folks want the plane information, they likely want the byte data.